compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/insert_cores.mlir

```diff
@@ -279,3 +279,104 @@ module {
+// CHECK:         amdaie.core
+// CHECK-COUNT-3: scf.for
+// CHECK-NOT:     scf.for
+// CHECK:         linalg.conv_1d_nwc_wcf
+// CHECK:         amdaie.end
+#map = affine_map<(d0) -> (d0 * 4)>
+#map1 = affine_map<(d0) -> (d0 * 8)>
+#translation = #iree_codegen.translation_info<Custom>
+builtin.module {
+  func.func @conv_2d_nhwc_hwcf_dispatch_0_conv_2d_nhwc_hwcf_2x12x12x64x3x3x32_i32() attributes {translation_info = #translation} {
+    %c768 = arith.constant 768 : index
+    %c9216 = arith.constant 9216 : index
+    %c16 = arith.constant 16 : index
+    %c128 = arith.constant 128 : index
+    %c384 = arith.constant 384 : index
+    %c192 = arith.constant 192 : index
+    %c1152 = arith.constant 1152 : index
+    %c64 = arith.constant 64 : index
+    %c2048 = arith.constant 2048 : index
+    %c6144 = arith.constant 6144 : index
+    %c4 = arith.constant 4 : index
+    %c448 = arith.constant 448 : index
+    %c6272 = arith.constant 6272 : index
+    %c6 = arith.constant 6 : index
+    %c2 = arith.constant 2 : index
+    %c1 = arith.constant 1 : index
+    %c32 = arith.constant 32 : index
+    %c3 = arith.constant 3 : index
+    %c0_i32 = arith.constant 0 : i32
+    %c0 = arith.constant 0 : index
+    %0 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%c0) flags(ReadOnly) : memref<2x14x14x32xi32>
+    %1 = amdaie.logicalobjectfifo.from_memref %0, {} : memref<2x14x14x32xi32> -> !amdaie.logicalobjectfifo<memref<2x14x14x32xi32>>
+    memref.assume_alignment %0, 64 : memref<2x14x14x32xi32>
+    %2 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) alignment(64) offset(%c0) flags(ReadOnly) : memref<3x3x32x64xi32>
+    %3 = amdaie.logicalobjectfifo.from_memref %2, {} : memref<3x3x32x64xi32> -> !amdaie.logicalobjectfifo<memref<3x3x32x64xi32>>
+    memref.assume_alignment %2, 64 : memref<3x3x32x64xi32>
+    %4 = hal.interface.binding.subspan set(0) binding(2) type(storage_buffer) alignment(64) offset(%c0) : memref<2x12x12x64xi32>
+    %5 = amdaie.logicalobjectfifo.from_memref %4, {} : memref<2x12x12x64xi32> -> !amdaie.logicalobjectfifo<memref<2x12x12x64xi32>>
+    memref.assume_alignment %4, 64 : memref<2x12x12x64xi32>
+    scf.forall (%arg0, %arg1, %arg2) in (3, 3, 16) {
+      %6 = affine.apply #map(%arg2)
+      %7 = affine.apply #map(%arg1)
+      %8 = affine.apply #map(%arg0)
+      %alloc = memref.alloc() : memref<2x6x6x32xi32, 1 : i32>
+      %9 = amdaie.logicalobjectfifo.from_memref %alloc, {} : memref<2x6x6x32xi32, 1 : i32> -> !amdaie.logicalobjectfifo<memref<2x6x6x32xi32, 1 : i32>>
+      %10 = amdaie.logicalobjectfifo.from_memref %alloc, {} : memref<2x6x6x32xi32, 1 : i32> -> !amdaie.logicalobjectfifo<memref<2x6x6x32xi32, 1 : i32>>
+      %11 = amdaie.dma_cpy_nd(%9[] [] [], %1[%c0, %8, %7, %c0] [%c2, %c6, %c6, %c32] [%c6272, %c448, %c32, %c1]) : (!amdaie.logicalobjectfifo<memref<2x6x6x32xi32, 1 : i32>>, !amdaie.logicalobjectfifo<memref<2x14x14x32xi32>>)
```

Comment:
This SSA isn't being used in the IR and the lit test can be made shorter. You may refer to either the [vectorization](https://github.com/nod-ai/iree-amd-aie/blob/0bf7b9f56b622844fb4cab8201664d2fdb0b4c43/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/insert_cores.mlir#L154) or [ukernel](https://github.com/nod-ai/iree-amd-aie/blob/0bf7b9f56b622844fb4cab8201664d2fdb0b4c43/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/insert_cores.mlir#L233) lit test and follow a similar structure.

Reply:
I agree, I'll strip this down a bit.